src/cargo/ops/cargo_compile/mod.rs

```diff
@@ -493,14 +497,36 @@ pub fn create_bcx<'a, 'gctx>(
                     continue;
                 };
 
-                warn_on_unexpected_cfgs(gctx, check_cfg, &cfg_expr, None, "")?;
+                warn_on_unexpected_cfgs(
+                    gctx,
+                    check_cfg,
+                    &cfg_expr,
+                    LintLevel::Warn,
+                    &mut error_count,
+                    None,
+                    "",
+                )?;
             }
 
             for unit in &units {
                 if !unit.show_warnings(gctx) {
                     continue;
                 }
 
+                let mut lint_level = LintLevel::Warn;
+                if let Ok(Some(lints)) = unit.pkg.manifest().normalized_toml().normalized_lints() {
+                    if let Some(rust_lints) = lints.get("rust") {
+                        if let Some(unexpected_cfgs) = rust_lints.get("unexpected_cfgs") {
+                            lint_level = unexpected_cfgs.level().into();
+                            max_lint_level = std::cmp::max(max_lint_level, lint_level);
+                        }
+                    }
+                }
+
+                if lint_level == LintLevel::Allow {
+                    continue;
+                }
+
```

Comment:
This incorrectly implements lint control in that its not subject to any groups which cargo currently has no knowledge of.

This leads into a more fundamental question we've been asking about lint control: when should we act on lints for `rust` / others vs use our own?

Examples
- unknown_lint
- snake_case, etc

This also ties into the question of how to handle clippy-like cargo lints, whether to be in cargo's namespace or clippy's.

Reply:
> This incorrectly implements lint control in that its not subject to any groups which cargo currently has no knowledge of.

Yeah, the special `warnings` group, didn't think of that.

> This leads into a more fundamental question we've been asking about lint control: when should we act on lints for rust / others vs use our own?

I also asked my-self the same question and I didn't wanted to introduce a cargo specific lint for it as it would mean duplicating the `[lints.rust.unexpected_cfgs.check-cfg]` entry for users.

-----

I don't know what the right answer is, but I feel like this could wait until there is a more general policy.